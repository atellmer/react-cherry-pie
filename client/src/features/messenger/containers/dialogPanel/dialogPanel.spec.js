import React from 'react';
import { shallow } from 'enzyme';
import { connect } from 'react-redux';

import { mountWithReduxAndContext } from 'test/mountWith';
import {
  DialogPanel,
  mapStateToProps,
  mapDispatchToProps
} from './dialogPanel';


describe('DialogPanel Container', () => {
  let props;

  beforeEach(() => {
    props = {
      isPhone: false,
      isTablet: false,
      isDesktop: false,
      widthWindow: 0,
      dialogItems: [],
      filterDialogs: jest.fn(),
      match: {},
      location: {}
    };
  });

  test('should render phone version of component', () => {
    const component = shallow(
      <DialogPanel
        {...props}
        isPhone='true' />
    );

    expect(component).toMatchSnapshot();
  });

  test('should render tablet(desktop) version of component', () => {
    const component = shallow(
      <DialogPanel
        {...props}
        isTablet='true' />
    );

    expect(component).toMatchSnapshot();
  });

  test('should render desktop and phone version of component', () => {
    let component;

    component = shallow(
      <DialogPanel
        {...props}
        isDesktop='true'
        widthWindow={1920} />
    );
    expect(component).toMatchSnapshot();

    component = shallow(
      <DialogPanel
        {...props}
        isDesktop='true'
        widthWindow={480} />
    );

    expect(component).toMatchSnapshot();
  });

  test('should pass props correctly', () => {
    let component;
    const expectedProps = {
      dialogItems: [],
      filterDialogs: jest.fn(),
      match: {},
      location: {}
    };

    component = shallow(
      <DialogPanel
        {...props}
        {...expectedProps}
        isPhone='true' />
    );
    expect(component.find('pure(DialogPanelPhone)').props()).toEqual(expectedProps);

    component = shallow(
      <DialogPanel
        {...props}
        {...expectedProps}
        isTablet='true' />
    );
    expect(component.find('pure(DialogPanelDesktop)').props()).toEqual(expectedProps);

    component = shallow(
      <DialogPanel
        {...props}
        {...expectedProps}
        isDesktop='true'
        widthWindow={480} />
    );
    expect(component.find('pure(DialogPanelPhone)').props()).toEqual(expectedProps);

    component = shallow(
      <DialogPanel
        {...props}
        {...expectedProps}
        isDesktop='true'
        widthWindow={1920} />
    );
    expect(component.find('pure(DialogPanelDesktop)').props()).toEqual(expectedProps);
  });

  test('should connect to redux correctly', () => {
    const expectedProps = {
      dialogItems: [],
      filterDialogs: expect.any(Function)
    };
    const DialogPanelWithState = connect(mapStateToProps, mapDispatchToProps)(DialogPanel);

    const component = mountWithReduxAndContext(
      <DialogPanelWithState />
    );

    expect(component.find(DialogPanel).props()).toEqual(expectedProps);
  });
});
