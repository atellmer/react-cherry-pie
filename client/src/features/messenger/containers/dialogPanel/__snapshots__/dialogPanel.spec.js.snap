// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DialogPanel Container should render desktop and phone version of component 1`] = `
<pure(DialogPanelDesktop)
  dialogItems={Array []}
  filterDialogs={[Function]}
  location={Object {}}
  match={Object {}}
/>
`;

exports[`DialogPanel Container should render desktop and phone version of component 2`] = `
<pure(DialogPanelPhone)
  dialogItems={Array []}
  filterDialogs={[Function]}
  location={Object {}}
  match={Object {}}
/>
`;

exports[`DialogPanel Container should render phone version of component 1`] = `
<pure(DialogPanelPhone)
  dialogItems={Array []}
  filterDialogs={[Function]}
  location={Object {}}
  match={Object {}}
/>
`;

exports[`DialogPanel Container should render tablet(desktop) version of component 1`] = `
<pure(DialogPanelDesktop)
  dialogItems={Array []}
  filterDialogs={[Function]}
  location={Object {}}
  match={Object {}}
/>
`;
